### PR TITLE
jxrlib: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jxrlib/all/conanfile.py [hunk 1]
```diff
@@ -23,60 +26,64 @@ class JxrlibConan(ConanFile):
...
+        except Exception:
+            pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
-        if self.settings.compiler == "Visual Studio" and self.options.shared:
-            raise ConanInvalidConfiguration("jxrlib shared not supported by Visual Studio")
+        if is_msvc(self) and self.options.shared:
```
Comment (jwillikers) [inline recipes/jxrlib/all/conanfile.py:56]: Is it possible to use `self.info.options.shared` here?

```suggestion
        if is_msvc(self) and self.info.options.shared:
```

**Review by jwillikers — CHANGES_REQUESTED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by prince-chrismc** [comment]
V2 pipeline is not passing 

Error running command `conan graph info --requires jxrlib/cci.20170615@#a23c9807b12818ff7a73fae89e58ce6a -f json  -pr:h /home/conan/w/prod-v2_cci_PR-13525/4/56f4d13e-dfb8-45e9-8631-ed5e0ac70f4a/profile_linux_11_libstdcpp11_gcc_release_64.-shared-False.txt -pr:b /home/conan/w/prod-v2_cci_PR-13525/4/56f4d13e-dfb8-45e9-8631-ed5e0ac70f4a/profile_linux_11_libstdcpp11_gcc_release_64.-shared-False.txt`:
   ```
   -------- Input profiles --------
   Profile host:
   [settings]
   arch=x86_64
   build_type=Release
   compiler=gcc
   compiler.libcxx=libstdc++11
   compiler.version=11
   os=Linux
   [options]
   */*:shared=False
   
   Profile build:
   [settings]
   arch=x86_64
   build_type=Release
   compiler=gcc
   compiler.libcxx=libstdc++11
   compiler.version=11
   os=Linux
   [options]
   */*:shared=False
   
   
   -------- Computing dependency graph --------
   Graph root
       virtual
   Requirements
       jxrlib/cci.20170615#a23c9807b12818ff7a73fae89e58ce6a - Cache
   
   -------- Computing necessary packages --------
   ERROR: jxrlib/cci.20170615: Error in validate() method, line 56
   	if is_msvc(self) and self.info.options.shared:
   	ConanException: 'self.settings' access in 'validate()' method is forbidden
   ```

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
> V2 pipeline is not passing
> 
> Error running command `conan graph info --requires jxrlib/cci.20170615@#a23c9807b12818ff7a73fae89e58ce6a -f json -pr:h /home/conan/w/prod-v2_cci_PR-13525/4/56f4d13e-dfb8-45e9-8631-ed5e0ac70f4a/profile_linux_11_libstdcpp11_gcc_release_64.-shared-False.txt -pr:b /home/conan/w/prod-v2_cci_PR-13525/4/56f4d13e-dfb8-45e9-8631-ed5e0ac70f4a/profile_linux_11_libstdcpp11_gcc_release_64.-shared-False.txt`:
> 
> ```
> -------- Input profiles --------
> Profile host:
> [settings]
> arch=x86_64
> build_type=Release
> compiler=gcc
> compiler.libcxx=libstdc++11
> compiler.version=11
> os=Linux
> [options]
> */*:shared=False
> 
> Profile build:
> [settings]
> arch=x86_64
> build_type=Release
> compiler=gcc
> compiler.libcxx=libstdc++11
> compiler.version=11
> os=Linux
> [options]
> */*:shared=False
> 
> 
> -------- Computing dependency graph --------
> Graph root
>     virtual
> Requirements
>     jxrlib/cci.20170615#a23c9807b12818ff7a73fae89e58ce6a - Cache
> 
> -------- Computing necessary packages --------
> ERROR: jxrlib/cci.20170615: Error in validate() method, line 56
> 	if is_msvc(self) and self.info.options.shared:
> 	ConanException: 'self.settings' access in 'validate()' method is forbidden
> ```

likely a conan v2 bug

**Comment by prince-chrismc** [comment]
> likely a conan v2 bug

The validate and self settings is the biggest reason why most recipes aren't passing that I know of

**Pull request merged**